### PR TITLE
[PW-5814] - Moving configuration functions from Data helper to config helper

### PR DESCRIPTION
**Pull request by raoulritter**
**Description**

This PR works in order to move the configuration functions from the Data helper to the configuration helper in order to improve the  

**Tested scenarios**

Fixes

### CODE REVIEW

File: Block/Info/AbstractInfo.php [hunk 1]
```diff
@@ -11,29 +11,32 @@
...
 
 class AbstractInfo extends \Magento\Payment\Block\Info
 {
+
     /**
-     * @var \Adyen\Payment\Helper\Data
+     * @var \Adyen\Payment\Helper\Config
      */
-    protected $_adyenHelper;
+    protected $_configHelper;
```
Comment (peterojo) [inline Block/Info/AbstractInfo.php:25]: Could you change this to `$configHelper`? Also in the rest of the PR

File: Helper/Data.php [hunk 2]
```diff
@@ -746,7 +635,7 @@ public function getLiveEndpointPrefix($storeId = null)
      */
     public function getCheckoutFrontendRegion($storeId = null)
```
Comment (peterojo) [inline Helper/Data.php:636]: Why not also move this function to the config helper?

File: Helper/Data.php [hunk 3]
```diff
@@ -1058,7 +947,7 @@ public function isPaymentMethodOfType(string $paymentMethod, string $type): bool
 
     public function getRatePayId()
     {
```
Comment (peterojo) [inline Helper/Data.php:949]: this could also be moved to config helper since it is only a getter

File: Helper/Data.php [hunk 4]
```diff
@@ -1016,10 +905,10 @@ public function getOneClickPaymentMethods($customerId, $storeId, $grandTotal, $s
         return $billingAgreements;
     }
 
-    public function isPerStoreBillingAgreement($storeId)
-    {
-        return !$this->getAdyenOneclickConfigDataFlag('share_billing_agreement', $storeId);
-    }
+//    public function isPerStoreBillingAgreement($storeId)
```
Comment (peterojo) [inline Helper/Data.php:908]: did you mean to delete this method?

File: Block/Info/AbstractInfo.php (hunk 1 above)
Comment (candemiralp) [reply Block/Info/AbstractInfo.php]: Part of this refactoring was done in #2105. Skipping this one to prevent confliction.

**Review by candemiralp — APPROVED** [review]

**Review by candemiralp — DISMISSED** [review]

**Review by peterojo — APPROVED** [review]

**Review by RokPopov — APPROVED** [review]

**Review by candemiralp — APPROVED** [review]

**Review by peterojo — APPROVED** [review]

**Review by peterojo — APPROVED** [review]

**Pull request merged**